src/stream.c

```diff
@@ -295,8 +295,12 @@ static void rtp_handler(const struct sa *src, const struct rtp_header *hdr,
 		flush = true;
 	}
 
-	if (s->jbuf) {
+	/* payload-type changed? */
+	err = s->pth(hdr->pt, mb, s->arg);
+	if (err)
+		return;
```

Comment:
I think that the pth handler should only be called, if the pt has changed.
something similar to this logic:

```
	/* Audio payload-type changed? */
	if (hdr->pt != rx->pt) {

		err = update_payload_type(a, rx->pt, hdr->pt);
		if (err)
			return;
	}
```

Reply:
ok.